services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,252 @@
+            {
+                this.Logger.LogDebug("The service is not running '{Status}'", this.State.Status);
+            }
+            else
+            {
+                try
+                {
+                    // TODO: Handle e-tag.
+                    // TODO: Update on some kind of interval.
+                    this.Lookups = await _api.GetLookupsAsync();
+
+                    // Listen to every enabled data source with a topic.
+                    var topics = !String.IsNullOrWhiteSpace(_options.Topics) ? _options.GetTopics() : this.Lookups?.DataSources
+                        .Where(ds => ds.IsEnabled &&
+                            ds.ContentTypeId > 0 &&
+                            !String.IsNullOrWhiteSpace(ds.Topic) &&
+                            ds.Connection.ContainsKey("import") &&
+                            ((JsonElement)ds.Connection["import"]).GetBoolean()).Select(ds => ds.Topic).ToArray() ?? Array.Empty<string>();
+
+                    if (topics.Length != 0)
+                    {
+                        var tps = String.Join(',', topics);
+                        this.Logger.LogInformation("Consuming topics: {tps}", tps);
+
+                        // TODO: Need to learn how to safely stop listening without losing content.  Every time a service goes down it will most likely lose content.
+                        await this.Consumer.ListenAsync<string, SourceContent>(HandleMessageAsync, topics);
+
+                        // Successful run clears any errors.
+                        this.State.ResetFailures();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.Logger.LogError(ex, "Service had an unexpected failure.");
+                    this.State.RecordFailure();
+                }
+            }
+
+            // The delay ensures we don't have a run away thread.
+            this.Logger.LogDebug("Service sleeping for {delay} ms", delay);
+            await Task.Delay(delay);
+        }
+    }
+
+    /// <summary>
+    /// Import the content.
+    /// Copy any file associated with source content.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> result)
+    {
+        if (this.State.Status != ServiceStatus.Running)
+        {
+            this.Consumer.Stop();
+            this.State.Stop();
+        }
+
+        this.Logger.LogInformation("Importing Content from Topic: {Topic}, Uid: {Key}", result.Topic, result.Message.Key);
+
+        // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
+        // TODO: Handle e-tag.
+        var source = await _api.GetDataSourceAsync(result.Value.Source) ?? throw new InvalidOperationException($"Failed to fetch data source for '{result.Topic}'");
```

Comment:
I don't think you need to make a request for the data source.  I believe you only need a request to fetch the content.

Reply:
So the values in the Content record:

            ContentTypeId = source.ContentTypeId.Value,
            MediaTypeId = source.MediaTypeId,
            LicenseId = source.LicenseId,
            OwnerId = source.OwnerId,
            DataSourceId = source.Id,

should be left undefined? I also use source.GetConnectionValue("serviceType") to determine where to look for audio. As I mentioned in a TODO comment, we won't want to do a transcription of a 24 hour long captured stream, so maybe we just need to look in clips.

When you mention "fetch the content" do you mean from Kafka? I'm assuming a record will not exist in the content table until I create one, but maybe I'm missing something?

Reply:
You're updating content.  The content will have all this information.  You don't want to override it, as you may be changing something a user has done.

Reply:
This has been resolved. I'm no longer reading from the Data Source but only from the Content.